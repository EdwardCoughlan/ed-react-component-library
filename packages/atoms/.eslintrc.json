{
  "extends": [
    "plugin:@nrwl/nx/react",
    "../../.eslintrc.json",
    "airbnb",
    "airbnb-typescript"
  ],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": ["packages/atoms/tsconfig.*?.json"]
      },
      "rules": {
        "react/react-in-jsx-scope" : "off"
      }
    },
    {
      "files": ["*.spec.ts", "*.spec.tsx"],
      "rules": {
        "import/no-extraneous-dependencies": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {
        "@typescript-eslint/explicit-function-return-type": ["error"]
      }
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
