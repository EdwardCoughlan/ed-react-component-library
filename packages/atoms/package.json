{
  "name": "@edwardcoughlan/atoms",
  "version": "0.0.1",
  "peerDependencies": {
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "classnames": "^2.3.1"
  }
}
