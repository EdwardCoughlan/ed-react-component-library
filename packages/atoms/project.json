{
  "sourceRoot": "packages/atoms/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/atoms",
        "tsConfig": "packages/atoms/tsconfig.lib.json",
        "project": "packages/atoms/package.json",
        "entryFile": "packages/atoms/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "assets": [
          {
            "glob": "packages/atoms/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/atoms/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/atoms"],
      "options": {
        "jestConfig": "packages/atoms/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
