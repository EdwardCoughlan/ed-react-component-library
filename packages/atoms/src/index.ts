export { default as Atoms } from './lib/atoms';
export * from './lib/atoms';
