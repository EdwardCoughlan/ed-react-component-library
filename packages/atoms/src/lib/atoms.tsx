import './atoms.module.scss';

/* eslint-disable-next-line */
export interface AtomsProps {
  welcomeMessage?: string;
}

const Atoms = ({
  welcomeMessage = 'Welcome to Atoms!',
}: AtomsProps): JSX.Element => (
  <div>
    <h1>${welcomeMessage}</h1>
  </div>
);

export default Atoms;
