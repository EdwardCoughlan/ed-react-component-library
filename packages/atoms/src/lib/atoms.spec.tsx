import { render } from '@testing-library/react';

import { Atoms } from './atoms';

describe('Atoms', () => {
  it('should render successfully', () => {
    const { baseElement, container } = render(<Atoms />);
    expect(baseElement).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
