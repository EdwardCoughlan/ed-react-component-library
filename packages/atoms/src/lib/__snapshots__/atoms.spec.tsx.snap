// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms should render successfully 1`] = `
<div>
  <div
    class="background"
  >
    <h1>
      Welcome to Atoms!
    </h1>
  </div>
</div>
`;
